lib/serve/index.ts

```diff
@@ -76,6 +84,57 @@ export class LisaServeApplicationStack extends Stack {
             vpc: vpc.vpc,
         });
 
+        const managementKeySecret = new Secret(this, createCdkId([id, 'managementKeySecret']), {
+            secretName: `lisa_management_key_secret-${Date.now()}`, // pragma: allowlist secret`
+            description: 'This is a secret created with AWS CDK',
+            generateSecretString: {
+                excludePunctuation: true,
+                passwordLength: 16
+            },
+        });
+
+        // const commonLambdaLayer = LayerVersion.fromLayerVersionArn(
+        //     this,
+        //     'base-common-lambda-layer',
+        //     StringParameter.valueForStringParameter(this, `${config.deploymentPrefix}/layerVersion/common`),
+        // );
+
+        // const rotateManagementKeyLambdaId = createCdkId([id, 'RotateManagementKeyLambda'])
+        // const rotateManagementKeyLambda = new Function(this, rotateManagementKeyLambdaId, {
+        //     functionName: rotateManagementKeyLambdaId,
+        //     runtime: Runtime.PYTHON_3_12,
+        //     handler: 'management_key.rotate_management_key',
+        //     code: Code.fromAsset('./lambda/'),
+        //     timeout: Duration.minutes(1),
+        //     memorySize: 1024,
+        //     environment: {
+        //         MANAGEMENT_KEY_NAME: managementKeySecret.secretName
+        //     },
+        //     layers: [commonLambdaLayer],
+        //     vpc: props.vpc.vpc,
+        // });
+
+        // managementKeySecret.grantRead(rotateManagementKeyLambda);
+
+        // new RotationSchedule(this, createCdkId([id, 'RotateManagementRotationSchedule']), {
+        //     secret: managementKeySecret,
+        //     rotationLambda: rotateManagementKeyLambda,
+        //     automaticallyAfter: Duration.days(1), // Rotate every 30 days
+        //   });
+
+        // You can now use the `secret` variable to access the created secret
+        // For example, you might output the secret's ARN for reference
+        new CfnOutput(this, createCdkId([id, 'RotateManagementKey']), {
+            value: managementKeySecret.secretArn,
+            description: 'The ARN of the secret',
+        });
```

Comment:
is this necessary/used? or do we just use the reference in other parts of the cdk definition?

Reply:
I don't think this is used